angular/projects/admin-nrpti/src/app/communications/lng-map-info/lng-map-info.component.ts

```diff
@@ -0,0 +1,129 @@
+  };
+
+  constructor(
+    public router: Router,
+    private factoryService: FactoryService,
+    private loadingScreenService: LoadingScreenService,
+    private _changeDetectionRef: ChangeDetectorRef
+  ) { }
+
+  ngOnInit() {
+    this.currentSection = this.lngMapData[0];
+    this.buildForm();
+  }
+
+  checkRoles() {
+    if (this.factoryService.userInRole('sysadmin')) {
+      return true;
+    } else if (this.factoryService.userInLngRole()) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * When selecting a segment, update the form with it's corresponding data
+   *
+   * @param {object} event
+   * @returns
+   * @memberof LngMapInfoComponent
+   */
+  onSelectSegment(event) {
+    if (!event) {
+      return;
+    }
+    const segment = event.target.value.split(': ');
+    this.currentSection = this.lngMapData.find(elem => elem.segment === segment[1]);
+    this.buildForm();
+    this._changeDetectionRef.detectChanges();
+  }
+
+  private buildForm() {
+    this.mapForm = new FormGroup({
+      sectionNumber: new FormControl({
+        value: (this.currentSection && this.currentSection.segment) || '',
+        disabled: !this.checkRoles()
+      }),
+      location: new FormControl({
+        value: (this.currentSection && this.currentSection.location) || '',
+        disabled: !this.checkRoles()
+      }),
+      length: new FormControl({
+        value: (this.currentSection && this.currentSection.length) || '',
+        disabled: !this.checkRoles()
+      }),
+      recentUpdates: new FormControl({
+        value: (this.currentSection && this.currentSection.description) || '',
+        disabled: !this.checkRoles()
+      })
+    });
+  }
+
+  async submit() {
+    this.loadingScreenService.setLoadingState(true, 'main');
+
+    if (
+      !this.mapForm.get('location').value ||
+      !this.mapForm.get('length').value ||
+      !this.mapForm.get('recentUpdates').value
+    ) {
+      alert('Please ensure your updates to the Section info are complete');
+    }
+
+    const mapInfo = {};
+    // currently we only allow edit on existing sections
+    mapInfo['_id'] = this.currentSection._id;
+    this.mapForm.controls.sectionNumber.dirty && (mapInfo['segment'] = this.mapForm.controls.sectionNumber.value);
+    this.mapForm.controls.location.dirty && (mapInfo['location'] = this.mapForm.controls.location.value);
+    this.mapForm.controls.length.dirty && (mapInfo['length'] = this.mapForm.controls.length.value);
+    this.mapForm.controls.recentUpdates.dirty && (mapInfo['description'] = this.mapForm.controls.recentUpdates.value);
+
+    const res = await this.factoryService.updateMapLayerInfo(mapInfo);
+    if (!res || !res._id) {
+      alert('Failed to update the LNG Section');
+    } else {
+      this.loadingScreenService.setLoadingState(false, 'main');
+    }
+
+    this.router.navigate(['communications', 'LNG']);
+  }
+
+  async cancel() {
+    // reset form values
+    this.buildForm();
```

Comment:
`currentSection` values aren't updated after a form submit.  This is causing the reset to use values from before the form changes.

To reproduce:

1. Select a section
2. Update the description text and save
3. Hit Cancel
4. Now the description is reset back to the value in Step 1.  The expected value should be value from Step 2

Reply:
Save should be triggering the resolver so that the description is update. I'll investigate this

Reply:
Save should not be triggering a resolver at all.  The message should popup and no angular router action should take place.